src/main/java/com/hillel/items_exchange/dto/SubcategoryDto.java

```diff
@@ -13,9 +16,15 @@
 @ToString
 public class SubcategoryDto {
 
+    @Zero(groups = New.class, message = "{invalid.new.entity.id}")
     @PositiveOrZero(message = "{invalid.id}")
```

Comment:
Unfortunately, I cannot do it because we have a method updateCategory in CategoryController with @Validated(Exist.class) for CategoryDto and this annotation applied to internal SubcategoryDtos. But in this method, SubcategoryDto can have id equals 0 (for a new entity) or has to exist by id.